presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/StaticMetastoreLocator.java

```diff
@@ -105,4 +135,38 @@ private static URI checkMetastoreUri(URI uri)
         checkArgument(uri.getPort() != -1, "metastoreUri port is missing: %s", uri);
         return uri;
     }
+
+    static class Backoff
+    {
+        private static final long MIN_BACKOFF = new Duration(50, MILLISECONDS).roundTo(NANOSECONDS);
+        private static final long MAX_BACKOFF = new Duration(60, SECONDS).roundTo(NANOSECONDS);
+
+        private final Ticker ticker;
+        private long lastFailureTimestamps;
+        private long backoffDuration;
+
+        Backoff()
+        {
+            backoffDuration = MIN_BACKOFF;
+            ticker = Ticker.systemTicker();
+        }
+
+        synchronized void fail()
+        {
+            lastFailureTimestamps = ticker.read();
+            backoffDuration = Math.min(backoffDuration * 2, MAX_BACKOFF);
+        }
+
+        synchronized void success()
+        {
+            lastFailureTimestamps = 0;
```

Comment:
Ticker may return `0`, `0` here is not a good idea for a missing value. Use `Optional` instead.

Reply:
Is readability your concern here?